src/Elastic.Apm/Agent.cs

```diff
@@ -162,16 +171,31 @@ private static bool CheckAndAddFilter(Action<PayloadSenderV2> action)
 
 		public static void Setup(AgentComponents agentComponents)
 		{
-			if (LazyApmAgent.IsValueCreated)
-				agentComponents?.Logger?.Error()
-					?.Log("The singleton APM agent has" +
-						" already been instantiated and can no longer be configured. Reusing existing instance");
-
-			agentComponents?.Logger?.Trace()
-				?.Log("Initialization - Agent.Setup called. Callstack: {callstack}", new StackTrace().ToString());
-
-			Components = agentComponents;
-			_isConfigured = true;
+			lock (InitializationLock)
+			{
+				if (_isConfigured)
+				{
+					Components?.Logger?.Error()
+						?.Log("The singleton APM agent has" +
```

Comment:
I tried, but then I realized this is a parametrized log. So we pass the callstack as a parameter to the `Log` method and that applies a formatter. This can then keep the parameters as key-value pairs (in this case `callstack` as key and then the callstack itself as value) and ship the log that way. E.g. in this case if you ship to elasticsearch this will be searchable.

If I prepare a string we loose the formatter which is applied by the logger.

No strong opinion, but after looking into this, I'd keep it this way.